base/strings/iteration.jl

```diff
@@ -0,0 +1,120 @@
+# A specialized iterator for EachIndex of strings
+struct EachStringIndex{T<:AbstractString}
+    s::T
```

Comment:
Somewhat tangential, but while we're moving this type, let's spell the field out as `.string` instead of `.s`.

Reply:
Can we have `nextindex` spelled out instead of `nextind` too? 🚲🏚